tensorflow/compiler/tf2tensorrt/convert/utils.cc

```diff
@@ -15,9 +15,328 @@ limitations under the License.
+  int to_dump = std::min((int)vals.NumElements(), max_len);
+  const T* val_vec = vals.flat<T>().data();
+  string msg =
+      StrCat(name, " , ", TYPESTR(T), "[", vals.NumElements(), "]", " v= ");
+  for (int f = 0; f < to_dump; f++) {
+    TypeDump(&msg, *(val_vec + f));
+  }
+  VLOG(0) << "Field " << msg;
+}
+#undef TYPESTR
+
+Status ConstructPlugin(const AttrSlice& attrs, const string& name,
+                       nvinfer1::IPluginV2*& plugin, bool validation_only) {
+  plugin = nullptr;
+
+  string plugin_name, plugin_version;
+  TF_RETURN_IF_ERROR(GetNodeAttr(attrs, "plugin_name", &plugin_name));
+  TF_RETURN_IF_ERROR(GetNodeAttr(attrs, "plugin_version", &plugin_version));
+  auto registry = getPluginRegistry();
+  if (!registry) {
+    return errors::Unavailable("Plugin registry is missing!");
+  }
+  auto creator =
+      registry->getPluginCreator(plugin_name.c_str(), plugin_version.c_str());
+  if (!creator) {
+    return errors::Unavailable("No plugin creator is registered for plugin '",
+                               plugin_name, "', version", plugin_version,
+                               ". Did you loaded your plugin library?");
+  }
+  const auto field_names = creator->getFieldNames();
+  std::vector<nvinfer1::PluginField> fields;
+  std::vector<Tensor> store;
+  // need special treatment of dims;
+  std::vector<nvinfer1::Dims> dim_store;
+
+  if (field_names) {
+    int num_fields = field_names->nbFields;
+    auto field = field_names->fields;
+    for (int i = 0; i < num_fields; ++i) {
+      if (!field->name) {
+        LOG(WARNING) << "Field " << i << " in plugin field map for " << name
+                     << " is nullptr";
+        field++;
+        continue;
+      }
+      VLOG(2) << "Trying to parse plugin requested attribute " << field->name;
+      string hidden_attr = StrCat("_", field->name);
+      const AttrValue* attr_value = nullptr;
+      Status attr_found=attrs.Find(hidden_attr, &attr_value);
+      if (!attr_found.ok()) {
+        LOG(WARNING) << "Field " << field->name
+                     << " is not defined in plugin node_def for " << name
+                     << " skipping field.";
+        field++;
+        continue;
+      }
+
+      const auto& attr_type = attr_value->value_case();
+      if (attr_type == AttrValue::kList) {
+        LOG(WARNING) << "Can't parse lists yet!";
+        field++;
+        continue;
+      }
+      if (attr_type != AttrValue::kTensor) {
+        LOG(WARNING) << field->name << " for op " << name
+                     << " is not a tensor. Need args to be encoded in tensors!";
+        field++;
+        continue;
+      }
+      Tensor t;
+      if (!t.FromProto(attr_value->tensor())) {
+        return errors::InvalidArgument("Can't parse tensor for op ", name,
+                                       " from attribute ", field->name);
+      }
+
+      switch (field->type) {
+        case nvinfer1::PluginFieldType::kFLOAT16: {
```

Comment:
Can we use macro like [these](https://github.com/tensorflow/tensorflow/search?q=define.*case&unscoped_q=define.*case)?